src/SkyCoords.jl

```diff
@@ -158,29 +158,32 @@ coords2cart(c::AbstractSkyCoords) = coords2cart(lon(c), lat(c))
 rotmat{T1<:GalCoords, T2<:ICRSCoords}(::Type{T1}, ::Type{T2}) = ICRS_TO_GAL
 rotmat{T1<:ICRSCoords, T2<:GalCoords}(::Type{T1}, ::Type{T2}) = GAL_TO_ICRS
 
+getequinox{eq}(x::Type{Type{FK5Coords{eq}}}) = eq
+getequinox{eq,T}(x::Type{Type{FK5Coords{eq,T}}}) = eq
+
 # Define both these so that `convert(FK5Coords{e}, ...)` and
 # `convert(FK5Coords{e,T}, ...)` both work. Similar with other
 # FK5Coords rotmat methods below.
-rotmat{e1, T2<:ICRSCoords}(::Type{FK5Coords{e1}}, ::Type{T2}) =
-    precess_from_j2000(e1) * ICRS_TO_FK5J2000
-rotmat{e1, T1, T2<:ICRSCoords}(::Type{FK5Coords{e1,T1}}, ::Type{T2}) =
-    precess_from_j2000(e1) * ICRS_TO_FK5J2000
-
-rotmat{e1, T2<:GalCoords}(::Type{FK5Coords{e1}}, ::Type{T2}) =
-    precess_from_j2000(e1) * GAL_TO_FK5J2000
-rotmat{e1, T1, T2<:GalCoords}(::Type{FK5Coords{e1,T1}}, ::Type{T2}) =
-    precess_from_j2000(e1) * GAL_TO_FK5J2000
-
-rotmat{T1<:ICRSCoords, e2, T2}(::Type{T1}, ::Type{FK5Coords{e2,T2}}) =
-    FK5J2000_TO_ICRS * precess_from_j2000(e2)'
-
-rotmat{T1<:GalCoords, e2, T2}(::Type{T1}, ::Type{FK5Coords{e2,T2}}) =
-    FK5J2000_TO_GAL * precess_from_j2000(e2)'
-
-rotmat{e1, e2, T2}(::Type{FK5Coords{e1}}, ::Type{FK5Coords{e2,T2}}) =
-    precess_from_j2000(e1) * precess_from_j2000(e2)'
-rotmat{e1, T1, e2, T2}(::Type{FK5Coords{e1,T1}}, ::Type{FK5Coords{e2,T2}}) =
-    precess_from_j2000(e1) * precess_from_j2000(e2)'
+@generated rotmat{E1,T2<:ICRSCoords}(e1::Type{FK5Coords{E1}}, ::Type{T2}) =
+    precess_from_j2000(getequinox(e1)) * ICRS_TO_FK5J2000
+@generated rotmat{E1,T1,T2<:ICRSCoords}(e1::Type{FK5Coords{E1,T1}}, ::Type{T2}) =
+    precess_from_j2000(getequinox(e1)) * ICRS_TO_FK5J2000
+
+@generated rotmat{E1,T2<:GalCoords}(e1::Type{FK5Coords{E1}}, ::Type{T2}) =
+    precess_from_j2000(getequinox(e1)) * GAL_TO_FK5J2000
+@generated rotmat{E1,T1,T2<:GalCoords}(e1::Type{FK5Coords{E1,T1}}, ::Type{T2}) =
+    precess_from_j2000(getequinox(e1)) * GAL_TO_FK5J2000
+
+@generated rotmat{T1<:ICRSCoords, E2, T2}(::Type{T1}, e2::Type{FK5Coords{E2, T2}}) =
+    FK5J2000_TO_ICRS * precess_from_j2000(getequinox(e2))'
+
+@generated rotmat{T1<:GalCoords, E2, T2}(::Type{T1}, e2::Type{FK5Coords{E2,T2}}) =
+    FK5J2000_TO_GAL * precess_from_j2000(getequinox(e2))'
+
+@generated rotmat{T1, E2, T2}(e1::Type{FK5Coords{T1}}, e2::Type{FK5Coords{E2,T2}}) =
```

Comment:
When `FK5Coords` has only one type parameter, isn't it the equinox? (Should this be `FK5Coords{E1}`?)

Reply:
Yes, even though the name doesn't matter here.  Will change.